modules/illumina.nf

```diff
@@ -102,10 +122,16 @@ process filterResidualAdapters {
     tuple val(sampleName), path(forward), path(reverse)
 
     output:
-    tuple val(sampleName), path("*1_posttrim_filter.fq.gz"), path("*2_posttrim_filter.fq.gz")
+    tuple val(sampleName), path("*1_posttrim_filter.fq.gz"), path("*2_posttrim_filter.fq.gz"), emit: filter_out
+    tuple val(sampleName), path("${sampleName}_filterResidualAdapters_provenance.yml"), emit: provenance
 
     script:
     """
+    printf -- "- process_name: filterResidualAdapters\\n"                                                 >> ${sampleName}_filterResidualAdapters_provenance.yml
+    printf -- "  tools:\\n"                                                                               >> ${sampleName}_filterResidualAdapters_provenance.yml
+    printf -- "    - tool_name: filter_residual_adapters.py\\n"                                           >> ${sampleName}_filterResidualAdapters_provenance.yml
+    printf -- "          sha256: \$(shasum -a 256 ${projectDir}/bin/filter_residual_adapters.py | cut -d ' ' -f 1)\\n"      >> ${sampleName}_filterResidualAdapters_provenance.yml
```

Comment:
I know this is a bit of a pain the way we implement writing this yaml using consecutive `printf` statements, but I think we need to be careful about whitespace and indentation to ensure that the output is a valid yaml file.

Could you adjust the indentation on the `sha256` line so the start of `sha256` aligns with the start of `tool_name` in the line above?

Reply:
fixed!